xcdat/temporal.py

```diff
@@ -2025,6 +2025,210 @@ def _calculate_departures(
         return ds_departs
 
 
+    def compute_monthly_average(self, data_var):
```

Comment:
This function wraps several steps/functions (that are defined below) in order to compute monthly averages (e.g., from hourly/daily/pentad data to monthly means):
* `ensure_bounds_order`: function ensures that dataset bounds are in order [earlier time, later time] (since PR logic depends on this)
* `generate_monthly_bounds`: function creates monthly bounds
* `get_temporal_weights`: function computes weights for averaging source dataset into targeted time periods
* `_experimental_averager`: function uses temporal weights to average data into targeted time periods

Reply:
I think we could generalize this function (kind of like `.temporal.group_average()`) by having it call different functions to generate target bounds (e.g., `generate_daily_bounds`, `generate_seasonal_bounds`, `generate_yearly_bounds`). The other steps would work as-is.